aws/resource_aws_dx_public_virtual_interface_test.go

```diff
@@ -61,77 +151,131 @@ func testAccCheckAwsDxPublicVirtualInterfaceDestroy(s *terraform.State) error {
 		if rs.Type != "aws_dx_public_virtual_interface" {
 			continue
 		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
 
-		input := &directconnect.DescribeVirtualInterfacesInput{
+		resp, err := conn.DescribeVirtualInterfaces(&directconnect.DescribeVirtualInterfacesInput{
 			VirtualInterfaceId: aws.String(rs.Primary.ID),
+		})
+		if isAWSErr(err, directconnect.ErrCodeClientException, "does not exist") {
+			continue
 		}
-
-		resp, err := conn.DescribeVirtualInterfaces(input)
 		if err != nil {
 			return err
 		}
-		for _, v := range resp.VirtualInterfaces {
-			if *v.VirtualInterfaceId == rs.Primary.ID && !(*v.VirtualInterfaceState == directconnect.VirtualInterfaceStateDeleted) {
-				return fmt.Errorf("[DESTROY ERROR] Dx Public VIF (%s) not deleted", rs.Primary.ID)
+
+		n := len(resp.VirtualInterfaces)
```

Comment:
Similar note here about preference to use `for` loop and potential for moving the lookup into a consistently used function.

Reply:
Same as [above](https://github.com/terraform-providers/terraform-provider-aws/pull/9572/files#r310260719).

Reply:
See comment [above](https://github.com/terraform-providers/terraform-provider-aws/pull/9572#discussion_r310261371).